RadioController.py

```diff
@@ -101,157 +109,302 @@ def parse_data(self, type_id, byte_list, length):
+    #     data = {}  # STUB
+    #     return data
 
-    def bulk_sensor(self, byte_list: List, length: int):
+    def bulk_sensor(self, byte_list: List, **kwargs):
         """
 
         :param byte_list:
         :type byte_list:
-        :param length:
-        :type length:
+        :param **kwargs:
+        :type **kwargs:
         :return:
         :rtype:
         """
         data: Dict[int, any] = {}
 
-        # TODO REVIEW/CHANGE in type refactoring: how this is required to convert from List[bytes] to List[int]
-        int_list: List[int] = [int(x[0]) for x in byte_list]
-
-        data[SubpacketEnum.TIME.value] = self.fourtoint(int_list[0:4])
-        data[SubpacketEnum.CALCULATED_ALTITUDE.value] = self.fourtofloat(int_list[4:8])  # TODO Double check it is calculated barometer altitude with firmware
-        data[SubpacketEnum.ACCELERATION_X.value] = self.fourtofloat(int_list[8:12])
-        data[SubpacketEnum.ACCELERATION_Y.value] = self.fourtofloat(int_list[12:16])
-        data[SubpacketEnum.ACCELERATION_Z.value] = self.fourtofloat(int_list[16:20])
-        data[SubpacketEnum.ORIENTATION_1.value] = self.fourtofloat(int_list[20:24])
-        data[SubpacketEnum.ORIENTATION_2.value] = self.fourtofloat(int_list[24:28])
-        data[SubpacketEnum.ORIENTATION_3.value] = self.fourtofloat(int_list[28:32])
-        data[SubpacketEnum.LATITUDE.value] = self.fourtofloat(int_list[32:36])
-        data[SubpacketEnum.LONGITUDE.value] = self.fourtofloat(int_list[36:40])
-        data[SubpacketEnum.STATE.value] = int_list[40]
+        # # TODO REVIEW/CHANGE in type refactoring: how this is required to convert from List[bytes] to List[int]
+        # byte_list: List[int] = [int(x[0]) for x in byte_list]
+        curr_byte = Count(0, 4)
+
+        data[SubpacketEnum.CALCULATED_ALTITUDE.value] = self.fourtofloat(byte_list[curr_byte.curr():curr_byte.next(4)])  # TODO Double check it is calculated barometer altitude with firmware
+        data[SubpacketEnum.ACCELERATION_X.value] = self.fourtofloat(byte_list[curr_byte.curr():curr_byte.next(4)])
+        data[SubpacketEnum.ACCELERATION_Y.value] = self.fourtofloat(byte_list[curr_byte.curr():curr_byte.next(4)])
+        data[SubpacketEnum.ACCELERATION_Z.value] = self.fourtofloat(byte_list[curr_byte.curr():curr_byte.next(4)])
+        data[SubpacketEnum.ORIENTATION_1.value] = self.fourtofloat(byte_list[curr_byte.curr():curr_byte.next(4)])
+        data[SubpacketEnum.ORIENTATION_2.value] = self.fourtofloat(byte_list[curr_byte.curr():curr_byte.next(4)])
+        data[SubpacketEnum.ORIENTATION_3.value] = self.fourtofloat(byte_list[curr_byte.curr():curr_byte.next(4)])
+        data[SubpacketEnum.LATITUDE.value] = self.fourtofloat(byte_list[curr_byte.curr():curr_byte.next(4)])  # TODO Should lat+lon be doubles??
+        data[SubpacketEnum.LONGITUDE.value] = self.fourtofloat(byte_list[curr_byte.curr():curr_byte.next(4)])
+        data[SubpacketEnum.STATE.value] = byte_list[curr_byte.currAndInc(1)]
```

Comment:
Starting to think that we should be using [io.BytesIO](https://docs.python.org/3/library/io.html#io.BytesIO). Note that it inherits [io.BufferedIOBase](https://docs.python.org/3/library/io.html#io.BufferedIOBase), so it has all the cool stream methods like `read()`, `write()`, `peak()` etc.

This would simplify to:
```suggestion
        data[SubpacketEnum.CALCULATED_ALTITUDE.value] = self.fourtofloat(byte_list.read(4))  # TODO Double check it is calculated barometer altitude with firmware
        data[SubpacketEnum.ACCELERATION_X.value] = self.fourtofloat(byte_list.read(4))
        data[SubpacketEnum.ACCELERATION_Y.value] = self.fourtofloat(byte_list.read(4))
        data[SubpacketEnum.ACCELERATION_Z.value] = self.fourtofloat(byte_list.read(4))
        data[SubpacketEnum.ORIENTATION_1.value] = self.fourtofloat(byte_list.read(4))
        data[SubpacketEnum.ORIENTATION_2.value] = self.fourtofloat(byte_list.read(4))
        data[SubpacketEnum.ORIENTATION_3.value] = self.fourtofloat(byte_list.read(4))
        data[SubpacketEnum.LATITUDE.value] = self.fourtofloat(byte_list.read(4))  # TODO Should lat+lon be doubles??
        data[SubpacketEnum.LONGITUDE.value] = self.fourtofloat(byte_list.read(4))
        data[SubpacketEnum.STATE.value] = byte_list[curr_byte.read(1)]
```

And we wouldn't need the `Count` helper class either. Also wouldn't need to pass around the header length.

What do you think about it?